src/metabase/analytics/prometheus.clj

```diff
@@ -0,0 +1,248 @@
+  than \"iapetos_internal\""
+  []
+  [(collector/named {:namespace "metabase_application"
+                     :name      "jvm_gc"}
+                    (GarbageCollectorExports.))
+   (collector/named {:namespace "metabase_application"
+                     :name      "jvm_standard"}
+                    (StandardExports.))
+   (collector/named {:namespace "metabase_application"
+                     :name      "jvm_memory_pools"}
+                    (MemoryPoolsExports.))
+   (collector/named {:namespace "metabase_application"
+                     :name      "jvm_threads"}
+                    (ThreadExports.))])
+
+(defn- jetty-collectors
+  []
+  ;; when in dev you might not have a server setup
+  (when (server/instance)
+    [(collector/named {:namespace "metabase_webserver"
+                       :name      "jetty_stats"}
+                      (JettyStatisticsCollector. (.getHandler (server/instance))))]))
+
+(defn- setup-metrics!
+  "Instrument the application. Conditionally done when some setting is set. If [[prometheus-server-port]] is not set it
+  will throw."
+  [registry-name]
+  (log/info (trs "Starting prometheus metrics collector"))
+  (let [registry (prometheus/collector-registry registry-name)]
+    (apply prometheus/register registry
+           (concat (jvm-collectors)
+                   (jetty-collectors)
+                   [@c3p0-collector]))))
+
+(defn- start-web-server!
+  "Start the prometheus web-server. If [[prometheus-server-port]] is not set it will throw."
+  [port registry]
+  (log/info (trs "Starting prometheus metrics web-server on port {0}" (str port)))
+  (when-not port
+    (throw (ex-info (trs "Attempting to set up prometheus metrics web-server with no web-server port provided")
+                    {})))
+  (ring-jetty/run-jetty (-> (constantly {:status 200})
+                            (collector.ring/wrap-metrics registry {:path "/metrics"}))
+                        {:join?       false
+                         :port        port
+                         :max-threads 8}))
+
+;;; API: call [[setup!]] once, call [[shutdown!]] on shutdown
+
+(defn setup!
+  "Start the prometheus metric collector and web-server."
+  []
+  (let [port (prometheus-server-port)]
+    (when-not port
+      (throw (ex-info (trs "Attempting to set up prometheus metrics with no web-server port provided")
+                      {})))
+    (when-not system
+      (locking #'system
+        (when-not system
+          (let [sys (make-prometheus-system port "metabase-registry")]
+            (alter-var-root #'system (constantly sys))))))))
+
+(defn shutdown!
+  "Stop the prometheus metrics web-server if it is running."
+  []
+  (when system
+    (locking #'system
+     (try (stop-web-server system)
+          (alter-var-root #'system (constantly nil))
+          (log/info (trs "Prometheus web-server shut down"))
+          (catch Exception e
+            (log/warn e (trs "Error stopping prometheus web-server")))))))
+
+(comment
```

Comment:
Did you mean to keep this comment?

Reply:
I'll sometimes leave a comment that shows how to interact with the system a little bit. This one is instructive as a way to get a dump of the metrics without needing to hit the webserver.